spec/database_admin_spec.rb

```diff
@@ -18,294 +18,705 @@
+
+          expect(subject.uri).to         eq(example_uri)
+          expect(subject.task).to        eq("evm:db:backup:remote")
+          expect(subject.task_params).to eq(["--", {:uri => example_uri}])
+        end
+      end
+    end
+
+    describe "#ask_smb_file_options" do
+      let(:example_uri) { subject.sample_url('smb') }
+      let(:user)        { 'example.com/admin' }
+      let(:pass)        { 'supersecret' }
+      let(:uri_prompt)  { "Enter the location of the remote backup file\nExample: #{example_uri}" }
+      let(:user_prompt) { "Enter the username with access to this file.\nExample: 'mydomain.com/user'" }
+      let(:pass_prompt) { "Enter the password for #{user}" }
+      let(:errmsg)      { "a valid URI" }
+
+      let(:expected_task_params) do
+        [
+          "--",
+          {
+            :uri          => example_uri,
+            :uri_username => user,
+            :uri_password => pass
+          }
+        ]
       end
 
-      it "sets @delete_agree to true if the user agrees" do
-        say "y"
-        subject.ask_to_delete_backup_after_restore
-        expect(subject.delete_agree).to eq(true)
+      context "with a valid uri, username, and password given" do
+        before do
+          say [example_uri, user, pass]
+          subject.ask_smb_file_options
+        end
+
+        it "sets @uri to point to the smb file" do
+          expect(subject.uri).to eq(example_uri)
+        end
+
+        it "sets @task to point to 'evm:db:backup:local'" do
+          expect(subject.task).to eq("evm:db:backup:remote")
+        end
+
+        it "sets @task_params to point to the smb file, username, and password" do
+          expect(subject.task_params).to eq(expected_task_params)
+        end
       end
 
-      it "sets @delete_agree to false if the user disagrees" do
-        say "n"
-        subject.ask_to_delete_backup_after_restore
-        expect(subject.delete_agree).to eq(false)
+      context "with a invalid uri given" do
+        let(:bad_uri) { "nfs://host.mydomain.com/path/to/file" }
+
+        before do
+          say [bad_uri, example_uri, user, pass]
+          subject.ask_smb_file_options
+        end
+
+        it "reprompts the user and then properly sets the options" do
+          error = "Please provide #{errmsg}"
+
+          expect_readline_question_asked uri_prompt
+          expect_readline_question_asked user_prompt
+          expect_heard [
+            uri_prompt,
+            error,
+            prompt,
+            "#{pass_prompt}: ***********\n"
+          ]
+
+          expect(subject.uri).to         eq(example_uri)
+          expect(subject.task).to        eq("evm:db:backup:remote")
+          expect(subject.task_params).to eq(expected_task_params)
+        end
       end
     end
 
-    context "when @backup_type not is LOCAL_FILE" do
-      let(:uri) { described_class::DB_RESTORE_FILE }
+    describe "#ask_to_delete_backup_after_restore" do
```

Comment:
Is this in the "for DB backup" context? I think this is restore related.

Reply:
Right, but because of how I have things, this get's fired every time.  That is why I am confirming a `no-op` with the spec below:

```ruby
it "no-ops" do
  # ...
end
```

Reply:
Yup it would help if I read the specs :sweat_smile: Carry on

Reply:
No worries, and I appreciate the attention to detail.  It also admittedly is a weird spec, so no shame in the mixup here.